test/hotspot/jtreg/compiler/c2/TestMergeLoads.java

```diff
@@ -0,0 +1,706 @@
+                (((long)(a[2] & 0xff)) << 16)|
+                (((long)(a[3] & 0xff)) << 24)|
+                (((long)(a[4] & 0xff)) << 32)|
+                (((long)(a[5] & 0xff)) << 40)|
+                (((long)(a[6] & 0xff)) << 48)|
+                (((long)(a[7] & 0xff)) << 56);
+      long[] ret = {i1};
+      return new Object[]{ret};
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_L_OF_CLASS, "byte\\\\[int:>=0] \\\\(java/lang/Cloneable,java/io/Serializable\\\\)", "1"},
+        applyIf = {"UseUnalignedAccesses", "true"})
+    static Object[] test2b(byte[] a) {
+      long i1 = UNSAFE.getLongUnaligned(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET, /* big-endian */ false);
+      long[] ret = {i1};
+      return new Object[]{ret};
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_L_OF_CLASS, "byte\\\\[int:>=0] \\\\(java/lang/Cloneable,java/io/Serializable\\\\)", "1"},
+        applyIf = {"UseUnalignedAccesses", "true"},
+        applyIfPlatformAnd = {"little-endian", "true"})
+    @IR(counts = {IRNode.LOAD_L_OF_CLASS, "byte\\\\[int:>=0] \\\\(java/lang/Cloneable,java/io/Serializable\\\\)", "1",
+                  IRNode.REVERSE_BYTES_L, "1"},
+        applyIf = {"UseUnalignedAccesses", "true"},
+        applyIfPlatform = {"big-endian", "true"})
+    static Object[] test2c(byte[] a) {
+      long i1 = ((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 0) & 0xff)       )|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 1) & 0xff)) << 8 )|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 2) & 0xff)) << 16)|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 3) & 0xff)) << 24)|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 4) & 0xff)) << 32)|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 5) & 0xff)) << 40)|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 6) & 0xff)) << 48)|
+               (((long)(UNSAFE.getByte(a, UNSAFE.ARRAY_BYTE_BASE_OFFSET + 7) & 0xff)) << 56);
+      long[] ret = {i1};
+      return new Object[]{ret};
+    }
+
+    /**
+     * Group 3: get int in big endian mode
+     */
+    @DontCompile
+    static Object[] test3R(byte[] a) {
+      int i1 = ((a[0] & 0xff) << 24) |
+               ((a[1] & 0xff) << 16) |
+               ((a[2] & 0xff) <<  8) |
+                (a[3] & 0xff);
+      int[] ret = {i1};
+      return new Object[]{ret};
+    }
+
+    @Test
+    @IR(counts = {IRNode.LOAD_I_OF_CLASS, "byte\\\\[int:>=0] \\\\(java/lang/Cloneable,java/io/Serializable\\\\)", "1",
+                  IRNode.REVERSE_BYTES_I, "1"},
+        applyIf = {"UseUnalignedAccesses", "true"},
+        applyIfPlatformAnd = {"little-endian", "true"})
+    @IR(counts = {IRNode.LOAD_I_OF_CLASS, "byte\\\\[int:>=0] \\\\(java/lang/Cloneable,java/io/Serializable\\\\)", "1"},
+        applyIf = {"UseUnalignedAccesses", "true"},
+        applyIfPlatform = {"big-endian", "true"})
```

Comment:
Can you please also check for byte and char loads? It would make sure that we do not have any loads that we are not expecting, and that the graph was cleaned appropriately.

Reply:
The other Load type are added in ir rules.